canonical/charmed-kubeflow-uats
# feat: Configure driver to support GPU tests

Implement changes as speced in the [spec](https://docs.google.com/document/d/1ptk_noideLk2m8Fntfh5wxpzQymFhLccKTKIqNIKIpc/edit?tab=t.0).
TDLR:
* Split tests to `cpu` and `gpu`
* Add `--include-gpu-tests` flag to include the tests under the `gpu` directory.
* Pass this flag to the tests job (together with the filter one) to include the appropriate tests.

Closes #132

### Testing
In order to test the PR
* Spin up a cluster with a GPU. For a two node microk8s node with one gpu, refer to this comment https://github.com/canonical/charmed-kubeflow-uats/issues/130#issuecomment-2465299876
* Run with the `--include-gpu-tests` flag e.g.
  ```
  tox -e uats-remote -- --include-gpu-tests --filter "placeholder"
  ```
  You should see only `placeholder` in the list (due the flag including it and filter making sure no other tests are run) and the job logs should include:
```
test_notebooks.py::test_notebook[placeholder] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:48 Note that only CPU tests will be run. In order to run tests that use an NVIDIA GPU,use the `--include-gpu-tests` flag e.g. `tox -e kubeflow-local -- --include-gpu-tests`.To learn more, use `--help` or refer to the repository's README file.
INFO     test_notebooks:test_notebooks.py:55 Running placeholder.ipynb...
PASSED                                                                   [100%]

======================= 1 passed, 9 deselected in 7.56s ========================
PASSED
```
  You could also run something without `--include-gpu-tests` in order to confirm that this PR does not affect running cpu UATs.
